external/vulkancts/modules/vulkan/wsi/vktWsiSwapchainTests.cpp

```diff
@@ -2417,6 +2418,81 @@ tcu::TestStatus destroyNullHandleSwapchainTest (Context& context, Type wsiType)
+
+	const auto fences = createFences(devHelper.vkd, *devHelper.device, numAcquirableImages + 1);
+	deUint32 dummy;
+	for (deUint32 i = 0; i < numAcquirableImages; ++i) {
+		VK_CHECK_WSI(devHelper.vkd.acquireNextImageKHR(*devHelper.device, *swapchain, std::numeric_limits<deUint64>::max(), (VkSemaphore)0, **fences[i], &dummy));
+	}
+
+	const auto result = devHelper.vkd.acquireNextImageKHR(*devHelper.device, *swapchain, 0, (VkSemaphore)0, **fences[numAcquirableImages], &dummy);
+
+	if (result != VK_SUCCESS && result != VK_SUBOPTIMAL_KHR && result != VK_NOT_READY ){
+		return tcu::TestStatus::fail("Implementation failed to respond well acquiring too many images with 0 timeout");
+	}
+
+	// cleanup
+	const deUint32 numFences = (result == VK_NOT_READY) ? static_cast<deUint32>(fences.size() - 1) : static_cast<deUint32>(fences.size());
+	vector<vk::VkFence> fencesRaw(numFences);
+	std::transform(fences.begin(), fences.begin() + numFences, fencesRaw.begin(), [](const FenceSp& f) -> vk::VkFence{ return **f; });
+	VK_CHECK(devHelper.vkd.waitForFences(*devHelper.device, numFences, fencesRaw.data(), VK_TRUE, std::numeric_limits<deUint64>::max()));
+
+	return tcu::TestStatus::pass("Acquire too many swapchain images test succeeded");
+}
+
+tcu::TestStatus acquireTooManyTimeoutTest (Context& context, Type wsiType)
+{
+	const tcu::UVec2               desiredSize     (256, 256);
+	const InstanceHelper           instHelper      (context, wsiType);
+	const NativeObjects            native          (context, instHelper.supportedExtensions, wsiType, tcu::just(desiredSize));
+	const Unique<VkSurfaceKHR>     surface         (createSurface(instHelper.vki, instHelper.instance, wsiType, *native.display, *native.window));
+	const DeviceHelper             devHelper       (context, instHelper.vki, instHelper.instance, *surface);
+	const VkSwapchainCreateInfoKHR swapchainInfo = getBasicSwapchainParameters(wsiType, instHelper.vki, devHelper.physicalDevice, *surface, desiredSize, 2);
+	const Unique<VkSwapchainKHR>   swapchain       (createSwapchainKHR(devHelper.vkd, *devHelper.device, &swapchainInfo));
+
+	deUint32 numImages;
+	VK_CHECK(devHelper.vkd.getSwapchainImagesKHR(*devHelper.device, *swapchain, &numImages, DE_NULL));
+	const deUint32 minImageCount = getPhysicalDeviceSurfaceCapabilities(instHelper.vki, devHelper.physicalDevice, *surface).minImageCount;
+	if (numImages < minImageCount) return tcu::TestStatus::fail("Get swapchain images returned less than minImageCount images");
+	const deUint32 numAcquirableImages = numImages - minImageCount + 1;
+
+	const auto fences = createFences(devHelper.vkd, *devHelper.device, numAcquirableImages + 1);
+	deUint32 dummy;
+	for (deUint32 i = 0; i < numAcquirableImages; ++i) {
+		VK_CHECK_WSI(devHelper.vkd.acquireNextImageKHR(*devHelper.device, *swapchain, std::numeric_limits<deUint64>::max(), (VkSemaphore)0, **fences[i], &dummy));
+	}
+
+	const deUint64 millisecond = 1000000;
```

Comment:
what is the reason behind this particular timeout?

Reply:
As the name of the test indicates, this needs (non-0) timeout.

Any arbitrary value will do. This one felt esthetically pleasing and realistic to me. More would be unnecessary waste of time, And otherwise waiting in nanosecond scale feels silly.

Reply:
It is not clear what are you trying to test with this case

Reply:
I test that driver acts in a conformant way when acquiring more than usual images-minImages+1 swapchain images, which is technically valid usage. This currently crashes NV, as https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/1363 says.

One test has 0 timeout, other has non-0 timeout. Those cases difer slightly per spec (mostly the expected `VkResult`).

Reply:
@alegal-arm I added a comment to the timeout if it helps.